feature/home/src/main/java/com/epmedu/animeal/home/presentation/ui/FeedSpotSheet.kt

```diff
@@ -0,0 +1,265 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Divider
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Favorite
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import com.epmedu.animeal.foundation.theme.CustomColor
+import com.epmedu.animeal.home.data.model.FeedSpot
+import com.epmedu.animeal.home.data.model.FeedStatus
+import com.epmedu.animeal.resources.R
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun FeedSpotSheetContent(
+    feedSpot: FeedSpot,
+    contentAlpha: Float,
+    onFavouriteChange: (Boolean) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .background(MaterialTheme.colors.surface)
+            .padding(
+                top = 8.dp,
+                bottom = 110.dp,
+                start = 24.dp,
+                end = 24.dp,
+            ),
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        Divider(
+            modifier = Modifier
+                .width(18.dp)
+                .clip(RoundedCornerShape(2.dp)),
+            thickness = 4.dp,
+        )
+        FeedSpotSheetHeader(
+            title = feedSpot.title,
+            status = feedSpot.status,
+            isFavourite = feedSpot.isFavourite,
+            onFavouriteChange = onFavouriteChange
+        )
+        FeedSpotDetails(
+            description = feedSpot.description,
+            lastFeederName = feedSpot.lastFeeder.name,
+            lastFeedTime = feedSpot.lastFeeder.time,
+            scrimAlpha = contentAlpha
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun HeartButton(
```

Comment:
I suggest splitting nested functions into separate files and cover with a preview.
also, I will be nice to have a preview of the whole `FeedSpotSheetContent`